airflow/utils/boolean.py

```diff
@@ -0,0 +1,31 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+def exactly_one(*args):
+    """
+    Returns True if exactly one of *args is "truthy", and False otherwise.
+    :param args:
+    :return:
+    """
+    has_one = False
+    for arg in args:
+        if arg and has_one:
+            return False
+        elif arg:
+            has_one = True
+    return has_one
```

Comment:
```suggestion
    return functools.reduce(operator.xor, map(bool, args))
```

😁

Reply:
lol

Reply:
hmm so the that's the thing with that XOR, this fails:
```
>>> functools.reduce(operator.xor, map(bool, ['a', 'a', 'a']))
True
```

(same goes for True in place of 'a')

The issue is that `True ^ True ^ True is True`

Reply:
Oh yeah fair point - xor trick doesn't work for more than two args.

Reply:
BTW. @dstandish - you do realise that a ^ b ^ c  is (one_or_three() ) :) ?

Reply:
> BTW. @dstandish - you do realise that a ^ b ^ c is (one_or_three() ) :) ?

yeah it's "exactly one OR all three" -- precisely the problem that my very old-fashioned and not-so-new-fangled code addresses :)

Reply:
like in the 3 case you can do `(a ^ b ^ c) and not (a and b and c)`... however, this is not generally true -- indeed it fails when you have 4 elements!